### PR TITLE
add a basic QUIC spec

### PR DESCRIPTION
**Pull request by marten-seemann**
Nothing fancy here. Creating this PR mostly to have a basis for the changes we’re planning for #498. Would appreciate getting this merged soon :)

### CODE REVIEW

File: quic/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# QUIC in libp2p
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active | r1, 2022-12-30  |
```
Comment (mxinden) [inline quic/README.md:5]:
```suggestion
| 3A              | Recommendation | Active | r1, 2022-12-30  |
```

Given that this is widely deployed, `Recommendation` seems the most appropriate to me.

**Review by mxinden — COMMENTED** [review]
What do you think of referencing the TLS specification here?

**Comment by MarcoPolo** [comment]
At least the libp2p TLS extension and the ALPN need to be specified.

**Comment by marten-seemann** [comment]
@mxinden @MarcoPolo That makes sense. I updated the document.

**Review by MarcoPolo — APPROVED** [review]

**Review by mxinden — APPROVED** [review]
Thanks for the follow-up.

**Comment by mxinden** [comment]
Before merging, can you please reference the specification in the root `README.md`?

**Review by elenaf9 — APPROVED** [review]

**Comment by marten-seemann** [comment]
> Before merging, can you please reference the specification in the root README.md?

Done.

**Pull request merged**